plugins/redfish/fu-self-test.c

```diff
@@ -8,7 +8,50 @@
 
 #include <fwupd.h>
 
+#include "fu-context-private.h"
+#include "fu-device-private.h"
+#include "fu-plugin-private.h"
+
 #include "fu-redfish-common.h"
+#include "fu-redfish-network.h"
+
+typedef struct {
+	FuPlugin		*plugin;
+} FuTest;
+
+static void
+fu_test_self_init (FuTest *self)
+{
+	gboolean ret;
+	g_autoptr(FuContext) ctx = fu_context_new ();
+	g_autoptr(GError) error = NULL;
+	g_autofree gchar *pluginfn = NULL;
+
+	ret = fu_context_load_quirks (ctx,
+				      FU_QUIRKS_LOAD_FLAG_NO_CACHE |
+				      FU_QUIRKS_LOAD_FLAG_NO_VERIFY,
+				      &error);
+	g_assert_no_error (error);
+	g_assert (ret);
+
+	self->plugin = fu_plugin_new (ctx);
+	pluginfn = g_build_filename (PLUGINBUILDDIR,
+				     "libfu_plugin_redfish." G_MODULE_SUFFIX,
+				     NULL);
+	ret = fu_plugin_open (self->plugin, pluginfn, &error);
+	g_assert_no_error (error);
+	g_assert (ret);
+	ret = fu_plugin_runner_startup (self->plugin, &error);
```

Comment:
how do you make sure it tries to use the `redfish.py` to make the self test results reproducible even on real hardware and not let smbios tables from the hardware influence the test?  Perhaps need to override the SMBIOS tables to `/dev/null`?

Reply:
I'm setting FWUPD_REDFISH_SMBIOS_DATA to get a static SMBIOS table, no?

Reply:
Ah I missed that, yes that's good